src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -247,4 +251,55 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
   return result;
 }
 
+bool VerifyCrlSignature(X509_CRL* crl, X509* issuer) {
+  if (issuer == nullptr || crl == nullptr) {
+    return false;
+  }
+  EVP_PKEY* ikey = X509_get_pubkey(issuer);
+  if (ikey == nullptr) {
+    // Can't verify signature because we couldn't get the pubkey, fail the
+    // check.
+    EVP_PKEY_free(ikey);
+    return false;
+  }
+  int ret = X509_CRL_verify(crl, ikey);
```

Comment:
nit: Since `X509_CRL_verify` can return values other than 0 and 1, should we change this to the following?

```
bool ok = (X509_CRL_verify(crl, ikey) == 1);
return ok;
```

Reply:
Done